substrate/frame/support/src/traits/tokens/asset_ops/common_ops.rs

```diff
@@ -157,88 +157,180 @@ where
 		strategy: S,
 		update_value: S::UpdateValue<'_>,
 	) -> Result<S::Success, DispatchError> {
-		let id = M::fallible_convert(id.clone())?;
+		let id = M::convert(id.clone())?;
 
 		Op::update(&id, strategy, update_value)
 	}
 }
 impl<Id, M, S, Op> Destroy<S> for MapId<Id, Op::Id, M, Op>
 where
-	M: FallibleConvert<Id, Op::Id>,
+	M: Convert<Id, Result<Op::Id, DispatchError>>,
 	S: DestroyStrategy,
 	Op: Destroy<S>,
 	Self::Id: Clone,
 {
 	fn destroy(id: &Self::Id, strategy: S) -> Result<S::Success, DispatchError> {
-		let id = M::fallible_convert(id.clone())?;
+		let id = M::convert(id.clone())?;
 
 		Op::destroy(&id, strategy)
 	}
 }
 impl<Id, M, S, Op> Stash<S> for MapId<Id, Op::Id, M, Op>
 where
-	M: FallibleConvert<Id, Op::Id>,
+	M: Convert<Id, Result<Op::Id, DispatchError>>,
 	S: StashStrategy,
 	Op: Stash<S>,
 	Self::Id: Clone,
 {
 	fn stash(id: &Self::Id, strategy: S) -> Result<S::Success, DispatchError> {
-		let id = M::fallible_convert(id.clone())?;
+		let id = M::convert(id.clone())?;
 
 		Op::stash(&id, strategy)
 	}
 }
 impl<Id, M, S, Op> Restore<S> for MapId<Id, Op::Id, M, Op>
 where
-	M: FallibleConvert<Id, Op::Id>,
+	M: Convert<Id, Result<Op::Id, DispatchError>>,
 	S: RestoreStrategy,
 	Op: Restore<S>,
 	Self::Id: Clone,
 {
 	fn restore(id: &Self::Id, strategy: S) -> Result<S::Success, DispatchError> {
-		let id = M::fallible_convert(id.clone())?;
+		let id = M::convert(id.clone())?;
 
 		Op::restore(&id, strategy)
 	}
 }
 
-/// This adapter allows one to derive a [CreateStrategy] value from the ID derivation parameters
-/// from the [DeriveAndReportId].
+/// The `CombinedAssetOps` is a tool for combining
+/// different implementations of `Restore`, `Update`, and `Stash` operations.
 ///
-/// The instance will be created using the derived strategy.
-pub struct DeriveStrategyThenCreate<Strategy, DeriveCfg, CreateOp>(
-	PhantomData<(Strategy, DeriveCfg, CreateOp)>,
+/// All three operations must use the same `AssetDefinition::Id`.
+pub struct CombinedAssetOps<RestoreOp, UpdateOp, StashOp>(
```

Comment:
I'm not sure about the name `CombinedAssetOps` since it isn't clear what it combines...
Perhaps, we could use something more clumsy yet clearer? Like `RestoreUpdateStashOps`?

Reply:
I'd also choose `CombinedAssetOps` from these variants. Yeah, let's stick with that until other combining helper types appear (if it even happens :D)